### PR TITLE
girafSimulate testing

### PR DESCRIPTION
**Pull request by dasnellings**

### CODE REVIEW

File: cmd/girafSimulate/girafSimulate_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	"github.com/vertgenlab/gonomics/giraf"
+	"io/ioutil"
+	"testing"
+)
+
+func TestGirafSimulate(t *testing.T) {
+	graph := MakeTestGraph()
+	out, err := ioutil.TempFile("", "GirafSimulate_test")
+	exception.PanicOnErr(err)
+	out.Close()
```
Comment (craiglowe) [inline cmd/girafSimulate/girafSimulate_test.go:18]: Does this return an error?

File: genomeGraph/giraf.go [hunk 2]
```diff
@@ -115,7 +118,9 @@ func RandSomaticMutations(graph *GenomeGraph, reads []*giraf.Giraf, numSomaticSN
 						continue
 					}
 					readPos = NodePosToReadPos(graph, reads[j], nodeIdx, pos)
-
+					if int(readPos) >= len(reads[j].Seq) {
```
Comment (craiglowe) [inline genomeGraph/giraf.go:121]: What is this checking?  Is this fixing a bug?

**Review by craiglowe — COMMENTED** [review]
Question about a possible bug fix in the PR

File: cmd/girafSimulate/girafSimulate_test.go (hunk 1 above)
Comment (dasnellings) [reply cmd/girafSimulate/girafSimulate_test.go:18]: Only error is 'file already closed'. So checking is unnecessary IMO.

File: genomeGraph/giraf.go (hunk 2 above)
Comment (dasnellings) [reply genomeGraph/giraf.go:121]: Fixing a bug. The RandLocationFast on newline 107 does not guarentee to pull a pos within the read, so I was getting index out of range errors. This check ensures the pos is valid.

**Review by craiglowe — APPROVED** [review]
Great!

**Review by cshoben — APPROVED** [review]

**Review by rimangan — APPROVED** [review]

**Pull request merged**